src/ic-websocket-cdk/src/state.rs

```diff
@@ -283,7 +306,10 @@ pub(crate) fn remove_client(client_key: &ClientKey) {
     if let Some(registered_client) =
         REGISTERED_CLIENTS.with(|map| map.borrow_mut().remove(client_key))
     {
-        decrement_gateway_clients_count(&registered_client.gateway_principal);
+        decrement_gateway_clients_count(
+            &registered_client.gateway_principal,
+            close_reason.is_none(),
```

Comment:
Why if it is `Some` you do not remove the gateway once it has no more clients connected?

Reply:
I've implemented a different way to handle this: the gateways are put in a GATEWAYS_TO_REMOVE map if empty, and removed from the state only after a ack interval ms time.